src/ripple/app/main/GRPCServer.h

```diff
@@ -0,0 +1,241 @@
+        // server.
+        rpc::v1::XRPLedgerAPIService::AsyncService& service_;
+
+        // The producer-consumer queue for asynchronous server notifications.
+        grpc::ServerCompletionQueue& cq_;
+
+        // Context for the rpc, allowing to tweak aspects of it such as the use
+        // of compression, authentication, as well as to send metadata back to
+        // the client.
+        grpc::ServerContext ctx_;
+
+        // true if finished processing request
+        bool finished_;
+
+        Application& app_;
+
+        // mutex for signaling abort
+        std::mutex mut_;
+
+        // whether the call should be aborted, due to server shutdown
+        bool aborted_;
+
+        // What we get from the client.
+        Request request_;
+
+        // What we send back to the client.
+        Response reply_;
+
+        // The means to get back to the client.
+        grpc::ServerAsyncResponseWriter<Response> responder_;
+
+        // Function that creates a listener for specific request type
+        BindListener<Request, Response> bindListener_;
+
+        // Function that processes a request
+        Handler<Request, Response> handler_;
+
+        // Condition required for this RPC
+        RPC::Condition requiredCondition_;
+
+        // Load type for this RPC
+        Resource::Charge loadType_;
+
+    public:
+        virtual ~CallData() = default;
+
+        // Take in the "service" instance (in this case representing an
+        // asynchronous server) and the completion queue "cq" used for
+        // asynchronous communication with the gRPC runtime.
+        CallData(
+            rpc::v1::XRPLedgerAPIService::AsyncService& service,
+            grpc::ServerCompletionQueue& cq,
+            Application& app,
+            BindListener<Request, Response> bindListener,
+            Handler<Request, Response> handler,
+            RPC::Condition requiredCondition,
+            Resource::Charge loadType);
+
+        virtual void
+        process() override;
+
+        virtual bool
+        isFinished() override;
+
+        virtual void
+        abort() override;
+
+        std::shared_ptr<Processor>
+        clone() override;
+
+    private:
+        // process the request. Called inside the coroutine passed to JobQueue
+        void
+        process(std::shared_ptr<JobQueue::Coro> coro);
+
+        // return load type of this RPC
+        Resource::Charge
+        getLoadType();
+
+        // return the Role required for this RPC
+        // for now, we are only supporting RPC's that require Role::USER for
+        // gRPC
+        Role
+        getRole();
+
+        // register endpoint with ResourceManager and return usage
+        Resource::Consumer
+        getUsage();
+
+    };  // CallData
+
+};  // GRPCServerImpl
+
+class GRPCServer
+{
+public:
+    GRPCServer(Application& app) : impl_(app){};
+
+    void
+    run();
+
+    ~GRPCServer();
```

Comment:
Recommend adding:
```
    GRPCServer(const GRPCServer&) = delete;
    GRPCServer& operator=(const GRPCServer&) = delete;
```
purely for readability purposes.  There is no change in functionality here.